manual/manual/html_processing/src/process_api.ml

```diff
@@ -0,0 +1,380 @@
+          (sprintf "The OCaml %sAPI" config.title)
+        |> prepend_child body;
+    | None ->
+        if search then search_widget false |> prepend_child nav;
+        (* Add "general index" link to all other files *)
+        create_element "a" ~inner_text:"< General Index"
+          ~attributes:["href", "index.html"]
+        |> prepend_child nav in
+
+  (* Add version number *)
+  add_version_link nav (config.title ^ "API Version " ^ version) releases_url;
+
+  (* Add sidebar button for mobile navigation *)
+  add_sidebar_button body;
+  
+  (* Add logo *)
+  prepend_child header (logo_html
+                          ((if config.title = "" then "" else "../") ^
+                           (manual_page_url ^ "/index.html")));
+
+  dbg "Saving %s..." out;
+
+  (* Save new html file *)
+  let new_html = to_string soup in
+  write_file out new_html
+
+let process ?(overwrite=false) ~version config file out =
+  if overwrite || not (Sys.file_exists out)
+  then Ok (process ~version config file out)
+  else Error (sprintf "File %s already exists." out)
+
+let all_html_files config =
+  Sys.readdir config.src_dir |> Array.to_list
+  |> List.filter (fun s -> Filename.extension s = ".html")
+
+
+(* Generate the index.js file for searching with the quick search widget *)
+(* The idea is to parse the file "index_values.html" to extract, for each entry
+   of this index, the following information (list of 8 strings):
+
+   [Module name; href URL of the Module (in principle an html file);
+   Value name; href URL of the value;
+   short description (html format); short description in txt format;
+   type signature (html format); type signature in txt format]
+
+   The "txt format" versions are used for searching, the "html version"
+   for display.
+   The signature is not in the "index_values.html" file, we have to look for it
+   by following the value href.
+
+   The index_values.html file has the following structure:
+
+   (...)
+
+   <table>
+
+   (...)
+
+   <tr><td><a href="List.html#VALappend">append</a> [<a href="List.html">List</a>]</td>
+   <td><div class="info">
+   <p>Concatenate two lists.</p>
+
+   </div>
+   </td></tr>
+
+   (...)
+
+   </table>
+
+   (...)
+
+   So we need to visit "List.html#VALappend", which has the following structure:
+
+   <pre><span id="VALappend"><span class="keyword">val</span> append</span> : <code class="type">'a list -> 'a list -> 'a list</code></pre>
+
+   and we finally return
+
+   ["List"; "List.html";
+   "rev_append"; "List.html#VALrev_append";
+   "<div class=\"info\">  <p><code class=\"code\"><span class=\"constructor\">List</span>.rev_append&nbsp;l1&nbsp;l2</code> reverses <code class=\"code\">l1</code> and concatenates it to <code class=\"code\">l2</code>.</p> </div>"; "  List.rev_append\194\160l1\194\160l2 reverses l1 and concatenates it to l2. ";
+   "<code class=\"type\">'a list -&gt; 'a list -&gt; 'a list</code>"; "'a list -> 'a list -> 'a list"]
+
+*)
+
```

Comment:
I would propose to make the index generation a separated module.

Reply:
OK; this was actually the case in the original version, for the "fast" Index.make that we decided to remove.

Reply:
done